climada/hazard/test/test_tc_tracks.py

```diff
@@ -248,6 +248,35 @@ def test_ibtracs_correct_pass(self):
         self.assertAlmostEqual(tc_try.data[0].central_pressure.values[5], 1008, places=0)
         self.assertAlmostEqual(tc_try.data[0].central_pressure.values[-1], 1012, places=0)
 
+    def test_read_with_basin(self):
+        """Filter TCs by (genesis) basin."""
+        # South Atlantic (not usually a TC location at all)
+        tc_track = tc.TCTracks()
+        tc_track.read_ibtracs_netcdf(basin="SA")
+        self.assertEqual(tc_track.size, 3)
+
+        # the basin is not necessarily the genesis basin
+        tc_track = tc.TCTracks()
+        tc_track.read_ibtracs_netcdf(year_range=(1995, 1995), basin="SP", estimate_missing=True)
+        self.assertEqual(tc_track.size, 6)
+        self.assertEqual(tc_track.data[0].basin[0], 'SP')
+        self.assertEqual(tc_track.data[5].basin[0], 'SI')
+
+        # genesis in NI
+        tc_track = tc.TCTracks()
+        tc_track.read_ibtracs_netcdf(
+            year_range=(1994, 1994), genesis_basin="NI", estimate_missing=True)
+        self.assertEqual(tc_track.size, 5)
+        for tr in tc_track.data:
+            np.testing.assert_array_equal(tr.basin[0], "NI")
```

Comment:
Why assert array equal, arent's these two single strings?

Reply:
Well spotted, this was a copy-paste thing.